chia/data_layer/data_layer.py

```diff
@@ -102,33 +108,22 @@ async def batch_update(
         changelist: List[Dict[str, Any]],
         fee: uint64,
     ) -> TransactionRecord:
-        hint_keys_values = await self.data_store.get_keys_values_dict(tree_id)
-        for change in changelist:
-            if change["action"] == "insert":
-                key = change["key"]
-                value = change["value"]
-                reference_node_hash = change.get("reference_node_hash")
-                side = change.get("side")
-                if reference_node_hash or side:
-                    await self.data_store.insert(key, value, tree_id, reference_node_hash, side, hint_keys_values)
-                await self.data_store.autoinsert(key, value, tree_id, hint_keys_values)
+        async with self.db_wrapper.locked_transaction(lock=True):
```

Comment:
Is this the data store's db?  I don't think we should be interacting with that outside of the store class itself.

Reply:
I've created a new one for now, for each batch_update. What happens if we get another write transaction while working on this batch update? (for example downloading the data from subscription). Since now there are 2 DBWrappers (as opposed to using a single lock as before everywhere), would that wait until we finish this batch update or it will throw an exception?